Use display name in directory browser breadcrumbs

LakshyaBagani opened the pull request:
### Fixes #16765

### Testing done
Verified locally. The breadcrumb now correctly shows the display name instead of the internal name, which looks much better for jobs with custom display names.

### UI Screenshots

After the change

<img width="1850" height="971" alt="Screenshot from 2025-12-12 04-13-50" src="https://github.com/user-attachments/assets/b4c0abfe-75b8-4f8c-be55-27be5c592a8e" />


Before the change

<img width="1850" height="926" alt="image" src="https://github.com/user-attachments/assets/819eecfa-4f82-4c3e-84bb-54fc11decc5d" />

### Proposed changelog entries
* Use display name in directory browser breadcrumb.

### Proposed changelog category
* Improvement

### Proposed upgrade guidelines
N/A

### Submitter checklist
- [x] The issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see examples). Fill in the Proposed upgrade guidelines section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with @Restricted or have @since TODO Javadocs, as appropriate.
- [ ] New deprecations are annotated with @Deprecated(since = "TODO") or @Deprecated(forRemoval = true, since = "TODO"), if applicable.
- [x] UI changes do not introduce regressions when enforcing the current default rules of Content Security Policy Plugin. In particular, new or substantially changed JavaScript is not defined inline and does not call eval to ease future introduction of Content Security Policy (CSP) directives (see documentation).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

Desired reviewers
@timja @mawinter69 @daniel-beck @MarkEWaite

timja commented:
Please include before and after screenshots for any ui change

daniel-beck reviewed (changes requested):
Please remove unrelated changes from this PR.

LakshyaBagani committed:
Update dir.jelly logic for navigation

LakshyaBagani committed:
Fix the navigation

LakshyaBagani commented:
I’ve added the before screenshot as well.

LakshyaBagani commented:
Thanks @daniel-beck and @mawinter69 for the feedback.

I see the issue with checking request2.originalRequestURI — it would indeed break if a job was named "artifact".

Based on @mawinter69's suggestion, I'll update the approach to avoid the URL check entirely. Instead, I will check if it.owner.name is null or empty.
1 :- If it is empty, I'll default to ${%Root}.
2 :- Otherwise, I'll use it.owner.name (or it.owner.displayName).

I will also verify this behavior across:
1 :- Freestyle project workspaces
2 :- Job artifacts
3 :- Run artifacts
4 :- Pipeline run workspaces

I'll push the changes shortly.

LakshyaBagani committed:
Change as per requirement

LakshyaBagani commented:
We removed the URL check entirely. Instead, we updated the logic in dir.jelly  to:

1 :- Check if it.owner.name is empty.
2 :- If empty: Display "Root" (this fixes the Artifacts view).
3 :- If not empty: Display it.owner.displayName (this keeps Workspaces working correctly).

Is Further Work Required?
@daniel-beck  @mawinter69

mawinter69 reviewed (changes requested):
Please also update screenshots

LakshyaBagani committed:
Change jelly

LakshyaBagani commented:
I had changed the code as per the requirement and also added the latest screenshot @daniel-beck @mawinter69 

<img width="1852" height="953" alt="image" src="https://github.com/user-attachments/assets/26be5129-a524-44d4-a6ae-df082e45a319" />
<img width="1852" height="953" alt="image" src="https://github.com/user-attachments/assets/20259a5b-fc95-4cc9-a273-3c6236fad205" />

LakshyaBagani commented:
@daniel-beck @mawinter69  any update in the PR . Do this required any further change or not ?

mawinter69 commented:
Can you update the screenshots and the original PR description at the top to reflect the current state. It doesn't fit anymore to what was changed and what was tested

mawinter69 reviewed (approved):
Otherwise it looks good I think

timja reviewed (approved):
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

LakshyaBagani commented:
@timja  just following up on this PR since it is already approved. Let me know if anything else is needed from my side before merging.

LakshyaBagani commented:
> > Please restore the PR template making sure a changelog entry is included: [`master`/.github/PULL_REQUEST_TEMPLATE.md](https://github.com/jenkinsci/jenkins/blob/master/.github/PULL_REQUEST_TEMPLATE.md?rgh-link-date=2026-01-15T22%3A13%3A04Z)

I had changed the PR description .

timja commented:
> > > Please restore the PR template making sure a changelog entry is included: [`master`/.github/PULL_REQUEST_TEMPLATE.md](https://github.com/jenkinsci/jenkins/blob/master/.github/PULL_REQUEST_TEMPLATE.md?rgh-link-date=2026-01-17T22%3A21%3A22Z)
> 
> I had changed the PR description .

Not until 15 minutes ago. I hope you've just mistaken had and have.

MarkEWaite reviewed (approved):
I tested interactively and found no issues.  Thanks!
